src/Models/ModelInstantiator.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace PDGA\DataObjects\Models;
+
+use PDGA\DataObjects\Attributes\Column;
+
+use ReflectionProperty;
+
+class ModelInstantiator
+{
+    /**
+     * Converts an associate array to a validated instance of a data object class.
+     *
+     * @param array  $arr   an array of values corresponding to the public properties of the data object.
+     * @param string $class Fully-qualified class name of the data object.
+     *
+     * @return object
+     */
+    public function arrayToDataObject(
+        array  $arr,
+        string $class
+    ): object
+    {
+        $instance = new $class();
+
+        // TODO validate.
+
+        // Assign all public properties.
+        foreach (array_keys(get_class_vars($class)) as $property)
+        {
+            if (!isset($arr[$property]))
+            {
+                continue;
+            }
+
+            $instance->{$property} = $arr[$property];
+        }
+
+        return $instance;
+    }
+
+    /**
+     * Converts a data object to an associative array with keys matching database fields for the
+     * corresponding database model.
+     *
+     * @param object $data_object An instance of a hydrated data object.
+     *
+     * @throws \ReflectionException
+     *
+     * @return array
+     */
+    public function dataObjectToDatabaseModel(
+        object $data_object
+    ): array
```

Comment:
The spec indicated that `string $class` should be a parameter here, but I think we can get that value from `$data_object::class`.